### PR TITLE
Move ExporterFactory to "exporter" package

### PR DESCRIPTION
**Pull request by tigrannajaryan**
This is part 2 of 3 of the refactoring plan to eliminate factories.go
and move the content to corresponding packages (receiver, exporter, processor).

### CODE REVIEW

File: configv2/configv2.go [hunk 1]
```diff
@@ -248,7 +249,7 @@ func loadExporters(v *viper.Viper) (models.Exporters, error) {
 		}
 
 		// Find exporter factory based on "type" that we read from config source
-		factory := factories.GetExporterFactory(typeStr)
+		factory := exporter.GetExporterFactory(typeStr)
```
Comment (pjanotti) [inline configv2/configv2.go]: nit: a rename seems good: `exporter.GetFactory(typeStr)`

Likely a symmetrical change applies to receivers, etc.

File: exporter/factory_test.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	if err != nil {
+		t.Fatalf("cannot register factory")
+	}
+
+	if &f != GetExporterFactory(f.Type()) {
+		t.Fatalf("cannot find factory")
+	}
+
+	// Verify that attempt to register a factory with duplicate name panics
+	paniced := false
```
Comment (pjanotti) [inline exporter/factory_test.go]: nit: `panicked`

File: exporter/factory_test.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+// CreateMetricsExporter creates a metrics exporter based on this config.
+func (f *ExampleExporterFactory) CreateMetricsExporter(logger *zap.Logger, cfg models.Exporter) (consumer.MetricsConsumer, StopFunc, error) {
+	return nil, nil, nil
+}
+
+func TestRegisterExporterFactory(t *testing.T) {
+	f := ExampleExporterFactory{}
+	err := RegisterExporterFactory(&f)
+	if err != nil {
```
Comment (pjanotti) [inline exporter/factory_test.go]: q: why not `require.NotNil` (and related below)? I think we should make one of the two practices the standard on the repo. I personally don't have much of a preference, but I like your argument about `require` and `assert` making the tests shorter.

File: configv2/configv2.go (hunk 1 above)
Comment (tigrannajaryan) [reply configv2/configv2.go]: Yes, I wanted to do exactly this, but in a future PR. :-)

File: exporter/factory_test.go (hunk 3 above)
Comment (tigrannajaryan) [reply exporter/factory_test.go]: These are not new tests, I just moved them. They were written before testify was added to the codebase. I agree we should try to follow a standard practice, but I don't want to touch the tests in this PR that is about something else.

File: exporter/factory_test.go (hunk 2 above)
Comment (tigrannajaryan) [reply exporter/factory_test.go]: Fixed.

File: configv2/configv2.go (hunk 1 above)
Comment (pjanotti) [reply configv2/configv2.go]: 👍

File: exporter/factory_test.go (hunk 3 above)
Comment (pjanotti) [reply exporter/factory_test.go]: 👍

**Comment by tigrannajaryan** [comment]
@bogdandrutu @pjanotti @songy23 can I get a couple re-approvals on this? I've been trying to merge this for a few hours already.

**Comment by bogdandrutu** [comment]
You got trapped by the same problem :))

**Pull request merged**